### PR TITLE
Add ValidateUrl in String

### PR DESCRIPTION
**Pull request by Yatin-kathuria**
# Welcome to JavaScript community

[Open in Gitpod](https://gitpod.io/from-referrer/)&nbsp;[know more](https://www.gitpod.io/docs/pull-requests/)

### Describe your change:

* [x] Add an algorithm?
* [ ] Fix a bug or typo in an existing algorithm?
* [ ] Documentation change?

### Checklist:
* [x] I have read [CONTRIBUTING.md](https://github.com/TheAlgorithms/Javascript/blob/master/CONTRIBUTING.md).
* [x] This pull request is all my own work -- I have not plagiarized.
* [x] I know that pull requests will not be merged if they fail the automated tests.
* [x] This PR only changes one algorithm file.  To ease review, please open separate PRs for separate algorithms.
* [x] All new JavaScript files are placed inside an existing directory.
* [x] All filenames should use the UpperCamelCase (PascalCase) style.  There should be no spaces in filenames.
     **Example:**`UserProfile.js` is allowed but `userprofile.js`,`Userprofile.js`,`user-Profile.js`,`userProfile.js` are not
* [x] All new algorithms have a URL in its comments that points to Wikipedia or other similar explanation.
* [ ] If this pull request resolves one or more open issues then the commit message contains `Fixes: #{$ISSUE_NO}`.

### CODE REVIEW

File: String/ValidateUrl.js [hunk 1]
```diff
@@ -0,0 +1,14 @@
+/**
+ * @function ValidateURL
+ * @description validate the URL.
+ * @param {String} url - The input URL string
+ * @return {Boolean}
+ * @see [ValidateURL](https://www.w3resource.com/javascript-exercises/javascript-regexp-exercise-9.php)
+ */
+const validateURL = (url) => {
+  const URL_PATTERN = /^(?:(?:https?|ftp):\/\/)?(?:(?!(?:10|127)(?:\.\d{1,3}){3})(?!(?:169\.254|192\.168)(?:\.\d{1,3}){2})(?!172\.(?:1[6-9]|2\d|3[0-1])(?:\.\d{1,3}){2})(?:[1-9]\d?|1\d\d|2[01]\d|22[0-3])(?:\.(?:1?\d{1,2}|2[0-4]\d|25[0-5])){2}(?:\.(?:[1-9]\d?|1\d\d|2[0-4]\d|25[0-4]))|(?:(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)(?:\.(?:[a-z\u00a1-\uffff0-9]-*)*[a-z\u00a1-\uffff0-9]+)*(?:\.(?:[a-z\u00a1-\uffff]{2,})))(?::\d{2,5})?(?:\/\S*)?$/
```
Comment (raklaptudirm) [inline String/ValidateUrl.js:9]: The regular expression is very inefficient. Please try to find other implementations of this functionality.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: String/ValidateUrl.js (hunk 1 above)
Comment (raklaptudirm) [reply String/ValidateUrl.js:9]: For example:
```js
/(https?:\/\/(?:www\.|(?!www))[^\s.]+\.[^\s]{2,}|www\.[^\s]+\.[^\s]{2,})/gi
```

Comment (Yatin-kathuria) [reply String/ValidateUrl.js]: update the regex pattern

File: String/ValidateUrl.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
+/**
+ * @function ValidateURL
+ * @description validate the URL.
+ * @param {String} url - The input URL string
+ * @return {Boolean}
+ */
+const validateURL = (url) => {
+  const URL_PATTERN = /(https?:\/\/(?:www\.|(?!www))[^\s.]+\.[^\s]{2,}|www\.[^\s]+\.[^\s]{2,})/gi
```
Comment (raklaptudirm) [inline String/ValidateUrl.js:8]: This regex will validate any strings which contain a valid link, for example:
`"link: https://www.google.com"`. Make sure the whole string matches the regex.

**Review by raklaptudirm — CHANGES_REQUESTED** [review]

File: String/ValidateUrl.js (hunk 2 above)
Comment (Yatin-kathuria) [reply String/ValidateUrl.js:8]: yes, it is  `https://www.google.com` is a valid URL

Comment (Yatin-kathuria) [reply String/ValidateUrl.js]: Fix that issue

**Review by raklaptudirm — APPROVED** [review]

**Pull request merged**